samples/go/server/server.go

```diff
@@ -0,0 +1,205 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"a2a/models"
+)
+
+// TaskHandler is a function type that handles task processing
+type TaskHandler func(task *models.Task, message *models.Message) (*models.Task, error)
+
+// A2AServer represents an A2A server instance
+type A2AServer struct {
+	handler     TaskHandler
+	port        int
+	basePath    string
+	taskStore   map[string]*models.Task
+	taskHistory map[string][]*models.Message
+	mu          sync.RWMutex
+}
+
+// NewA2AServer creates a new A2A server instance
+func NewA2AServer(handler TaskHandler, port int, basePath string) *A2AServer {
```

Comment:
Simile to `samples/go/client/client.go`, it would be nice to have a helper method for instantiating an `A2AServer` with an `AgentCard`.